resource/docker/volume/volume.go

```diff
@@ -0,0 +1,165 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package volume
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/asteris-llc/converge/resource"
+	"github.com/asteris-llc/converge/resource/docker"
+	dc "github.com/fsouza/go-dockerclient"
+	"golang.org/x/net/context"
+)
+
+// State type for Volume
+type State string
+
+const (
+	// StatePresent indicates the volume should be present
+	StatePresent State = "present"
+
+	// StateAbsent indicates the volume should be absent
+	StateAbsent State = "absent"
+)
+
+// Volume is responsible for managing docker volumes
+type Volume struct {
+	*resource.Status
+	client docker.VolumeClient
+
+	Name    string
+	Labels  map[string]string
+	Driver  string
+	Options map[string]string
+	State   State
+	Force   bool
+}
+
+// Check system for docker volume
+func (v *Volume) Check(context.Context, resource.Renderer) (resource.TaskStatus, error) {
+	v.Status = resource.NewStatus()
+	vol, err := v.client.FindVolume(v.Name)
+
+	if err != nil {
+		v.Status.Level = resource.StatusFatal
+		return v, err
+	}
+
+	v.Status.AddDifference(v.Name, string(volumeState(vol)), string(v.State), "")
+
+	if v.State == StatePresent && vol != nil && v.Force {
+		expectedLabels := mapToString(v.Labels)
+		actualLabels := mapToString(vol.Labels)
+		v.Status.AddDifference("labels", actualLabels, expectedLabels, "")
```

Comment:
it's an option but this approach is consistent with the existing docker resources. this is discussed more in #325. let me know if you think I should change it.

we could also update the human printer to only print diffs with changes. something like:

```diff
diff --git a/prettyprinters/human/human.go b/prettyprinters/human/human.go
index 74d41ac..4e137a8 100644
--- a/prettyprinters/human/human.go
+++ b/prettyprinters/human/human.go
@@ -136,7 +136,9 @@ func (p *Printer) DrawNode(g *graph.Graph, id string) (pp.Renderable, error) {
 	Has Changes: {{if .HasChanges}}{{yellow "yes"}}{{else}}no{{end}}
 	Changes:
 		{{- range $key, $values := .Changes}}
+		{{- if $values.Changes}}
 		{{cyan $key}}:	{{diff ($values.Original) ($values.Current)}}
+		{{- end}}
 		{{- else}} No changes {{- end}}
 
 `)
```